webapp/tests/karma/ts/services/update-settings.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { expect } from 'chai';
+
+import { UpdateSettingsService } from '@mm-services/update-settings.service';
+
+describe('UpdateSettings service', () => {
+  let service: UpdateSettingsService;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    service = TestBed.inject(UpdateSettingsService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('updates settings', () => {
+    const updates = {
+      isTrue: true,
+      isString: 'hello'
+    };
+    service.update(updates).then((result: any) => {
+      expect(result.success).to.equal(true);
```

Comment:
you're calling this async function and then not waiting for it to execute before ending the test. This expression will evaluate AFTER the test is finished.